contracts/v2/arbitrum-orbit/FiatTokenArbitrumOrbitV2_2.sol

```diff
@@ -0,0 +1,128 @@
+     * This is the keccak-256 hash of "FiatTokenArbitrumOrbitV2_2.l2Gateway.slot" subtracted by 1.
+     * Slot is hard-coded so that base USDC contracts can be further expanded without the possibility
+     * of slot collision.
+     */
+    bytes32
+        private constant L2_GATEWAY_SLOT = 0xdbf6298cab77bb44ebfd5abb25ed2538c2a55f7404c47e83e6531361fba28c24;
+
+    /**
+     * @dev Storage slot with the address of the parent chain USDC
+     * This is the keccak-256 hash of "FiatTokenArbitrumOrbitV2_2.l1Address.slot" subtracted by 1.
+     * Slot is hard-coded so that base USDC contracts can be further expanded without the possibility
+     * of slot collision.
+     */
+    bytes32
+        private constant L1_ADDRESS_SLOT = 0x54352c0d7cc5793352a36344bfdcdcf68ba6258544ce1aed71f60a74d882c191;
+
+    modifier onlyGateway() {
+        require(msg.sender == l2Gateway(), "ONLY_GATEWAY");
+        _;
+    }
+
+    /**
+     * @notice initialize the token
+     * @param l2Gateway_ child chain gateway this token communicates with
+     * @param l1Counterpart_ parent chain address of the USDC
+     */
+    function initializeArbitrumOrbit(address l2Gateway_, address l1Counterpart_)
+        external
+    {
+        // initializeV2_2 of FiatTokenV2_2 sets version to 3
+        require(_initializedVersion == 3);
+        require(l2Gateway_ != address(0), "INVALID_GATEWAY");
+        require(l2Gateway() == address(0), "ALREADY_INIT");
+        assembly {
+            sstore(L2_GATEWAY_SLOT, l2Gateway_)
+            sstore(L1_ADDRESS_SLOT, l1Counterpart_)
+        }
+    }
+
+    /**
+     * @notice Get the address of the child chain gateway connected to this token
+     */
+    function l2Gateway() public view returns (address _l2Gateway) {
+        assembly {
+            _l2Gateway := sload(L2_GATEWAY_SLOT)
+        }
+    }
+
+    /**
+     * @notice Get the address of the parent chain USDC
+     */
+    function l1Address() public override view returns (address _l1Address) {
+        assembly {
+            _l1Address := sload(L1_ADDRESS_SLOT)
+        }
+    }
+
+    /**
+     * @notice Mint tokens on child chain. Callable path is L1Gateway depositToken (which handles L1 escrow), which triggers L2Gateway, which calls this
+     * @param account recipient of tokens
+     * @param amount amount of tokens minted
+     */
+    function bridgeMint(address account, uint256 amount)
+        external
+        virtual
+        override
+        onlyGateway
+    {
+        mint(account, amount);
+    }
+
+    /**
+     * @notice Burn tokens on child chain.
+     * @dev only the token bridge gateway can call this
+     * @param account owner of tokens
+     * @param amount amount of tokens burnt
+     */
+    function bridgeBurn(address account, uint256 amount)
+        external
+        virtual
+        override
+        onlyGateway
+    {
+        _burn(account, amount);
+    }
+
+    /**
+     * @notice Allows a minter to burn some of its own tokens.
```

Comment:
wrong comment as-is ~own tokens~

Reply:
Was copy-pasted from base contract, now left just explanation on why logic is duplicated